### ISSUE TITLE
Drive various event activity from data file

### ISSUE THREAD

**Issue by nickebbitt** [ManchesterJavaCommunity/manchesterjavacommunity.github.io#32]
I was thinking it would be nice if we could add some config to a file that described an event e.g.

events:
  - title: My Event
    date: 01/12/2021
    description: Lots of useful info
.
.
.
etc.

We could then use this to drive the Event page of the website but we could also use it to:
- Create the Meetup event via the API
- Create an announcement tweet
- Announce in other social media platforms 

This might be something we can automate using GitHub actions

**Comment by CraigMcDonaldCodes** [comment]
I've had a similar idea but it was really just a poor replacement for a database. I had only been thinking in terms of the website though, probably a JSON file that can be loaded? Not sure if there is an existing JS library that does it or if JBake can help with this.

**Comment by nickebbitt** [comment]
I'd argue that a document is a good starting point before you need a database. 

A database adds lots of complexity whereas a static JSON/YAML document provides your data model without the the additional concerns. In the main I can see the data being pretty static in nature so value of a database initially feels minimal.

@jonbullock can maybe comment on how/if we can load a structured data file and reference it during the static site generation

**Comment by jonbullock** [comment]
Yeah JBake supports data files in YAML format (JSON support shouldn't be that much work to support). The contents is parsed, saved into the content store and exposed to the template engines. It's quite a recent addition and used on the optaplanner.org website to list events.

**Comment by jonbullock** [comment]
Don't see a reason why the same file can't be used for both the web site and some GitHub Actions.
